cdap-security/src/main/java/io/cdap/cdap/security/encryption/guice/DefaultAeadCipherService.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.encryption.guice;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import io.cdap.cdap.security.encryption.AeadCipherService;
+import io.cdap.cdap.security.spi.encryption.AeadCipher;
+import io.cdap.cdap.security.spi.encryption.AeadCipherContext;
+import io.cdap.cdap.security.spi.encryption.CipherOperationException;
+import java.util.Map;
+
+/**
+ * Default implementation of {@link AeadCipherService} which delegates calls to an instance of
+ * {@link AeadCipher}.
+ */
+public class DefaultAeadCipherService extends AbstractIdleService implements AeadCipherService {
+
+  private final AeadCipher aeadCipher;
+  private final Map<String, String> properties;
+  private Map<String, String> secureProperties;
+
+  protected DefaultAeadCipherService(AeadCipher aeadCipher, Map<String, String> properties,
+      Map<String, String> secureProperties) {
+    this.aeadCipher = aeadCipher;
+    this.properties = properties;
+    this.secureProperties = secureProperties;
+  }
+
+
+  @Override
+  protected void startUp() throws Exception {
+    aeadCipher.initialize(new AeadCipherContext(properties, secureProperties));
```

Comment:
Does it need to be a Service or can it be just a one time on demand initialization? Using Service making the usage a little bit more involving.

Reply:
Changed to be lazily initialized instead of initialized as a service